civis/futures.py

```diff
@@ -221,6 +221,12 @@ def _poll_and_set_api_result(self):
             except Exception as e:
                 self._set_api_exception(exc=e)
 
+    def outputs(self):
+        """Block on job completion and return a list of run outputs."""
+        self.result()
```

Comment:
Could you use a `concurrent.futures.wait` instead? The `.result()` will raise if the job fails, but you could still have outputs on a failed job that you want to retrieve.

Reply:
Hmm... raising an exception if the job fails seems desirable, by default at least.  I think a user generally  wouldn't want outputs if the job failed. If they have some situation where that's not the case, they could still use `wait` and the runs_outputs endpoint.

🤷‍♂

Reply:
Erroring on a failed job is a reasonable thing to do here, but it's not what I personally would have expected from the function name and docstring. I have jobs which do produce useful output on failure (for example, a validation step in a workflow which runs Great Expectations and outputs table details whether it passes or fails, or jobs which write debug logs to run outputs).

I guess the most important thing is to document the behavior on job failure in the docstring. You could imagine a flag to select the behavior on failure, but maybe that's adding too much complexity.

Reply:
How about we continue calling `result()` here and adding a full docstring with a `Raises` section that explains that the particular exception will be raised when the job fails?

Reply:
That would work for me, thanks!